GUI/Dialogs/SettingsDialog.cs

```diff
@@ -77,25 +77,35 @@ private void UpdateRefreshRate()
 
         private void RefreshReposListBox()
         {
-            // Give the Repository the priority it
-            // currently has in the gui
-            for (int i = 0; i < _sortedRepos.Count; i++)
+            var manager = RegistryManager.Instance(Main.Instance.CurrentInstance);
+            var registry = manager.registry;
+
+            if (_sortedRepos.Count == 0)
```

Comment:
It seems we are asking this function to handle two different tasks (loading, saving) and making it guess which one to perform, whereas the functions that call it definitely know (`SettingsDialog_Load` &rarr; `UpdateDialog` &rarr; `RefreshReposListBox` definitely doesn't need to save, but `DeleteRepoButton_Click` &rarr; `RefreshReposListBox` always would need to save).

What do you think of splitting this into two functions and only calling the one we need? Or maybe using a parameter to communicate what it should do?

Reply:
Good point. I went with the parameter option, because it felt cleaner than separate functions (which would've needed some duplicated lines), and was easier with the changes I already had.